### PR TITLE
Don't show create work button to users who don't have the depositor role

### LINKED ISSUE #677: A new repo allows a user without any admin sets to deposit
This should have been fixed in #615, but it seems like it's not working.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #677

### CODE REVIEW

File: app/presenters/hyrax/homepage_presenter.rb [hunk 1]
```diff
@@ -6,8 +6,19 @@ def initialize(current_ability)
       @current_ability = current_ability
     end
 
+    # @return [Boolean] If the current user is a guest and the always_display_share_button
+    #   is activated, then return true. Otherwise return true if the signed in
+    #   user has permission to create at least one kind of work.
     def display_share_button?
-      Hyrax.config.always_display_share_button? || current_ability.can_create_any_work?
+      (user_unregistered? && Hyrax.config.always_display_share_button?) ||
```
Comment (jeremyf) [inline app/presenters/hyrax/homepage_presenter.rb]: I'm a bit confused by the logic, is it the same as the following? If so, I prefer the following as it flows better when reading. There is a nuanced difference in that the `Hyrax.config.always_display_share_button?` has a higher order of precedence.

The logic as it stands seems to say only care about the `always_display_share_button?` if we have an unregistered user.

```ruby
return true if Hyrax.config.always_display_share_button?
return false if user_unregistered?
current_ability.can_create_any_work?
```

Comment (jcoyne) [inline app/presenters/hyrax/homepage_presenter.rb]: No, it is not the same.  `Hyrax.config.always_display_share_button?` is only a factor if the user is not logged in.

Comment (jeremyf) [inline app/presenters/hyrax/homepage_presenter.rb]: Is that feature a bit of a misnomer then? I read "always_display_share_button?" as if this is true then method "display_share_button?" should also be true.

Comment (jcoyne) [inline app/presenters/hyrax/homepage_presenter.rb]: It was a good name once upon a time, when all registered users could create all work types. It's no longer a good name.  Want me to change that now?

Comment (jeremyf) [inline app/presenters/hyrax/homepage_presenter.rb]: If you could change it that would help clear up what this means. Is that a breaking change?

Comment (jcoyne) [inline app/presenters/hyrax/homepage_presenter.rb]: We could probably do it with some deprecation in the configuration.

File: lib/hyrax/configuration.rb [hunk 2]
```diff
@@ -365,14 +365,21 @@ def cache_path
...
-    deprecation_deprecate always_display_share_button: "use always_display_share_button? instead"
+    # Should a button with "Share my work" show on the front page to users who are not logged in?
+    attr_writer :display_share_button_when_not_logged_in
+    def display_share_button_when_not_logged_in?
+      return true if @display_share_button_when_not_logged_in.nil?
+      @display_share_button_when_not_logged_in
+    end
+
+    alias always_display_share_button= display_share_button_when_not_logged_in=
+    deprecation_deprecate :always_display_share_button= => "use always_display_share_button= instead"
```
Comment (jeremyf) [inline lib/hyrax/configuration.rb]: Should this say:

```ruby
deprecation_deprecate :always_display_share_button= => "use display_share_button_when_not_logged_in = instead"
```

Comment (jcoyne) [inline lib/hyrax/configuration.rb]: Yep. Change made.

**Pull request merged**